### ISSUE TITLE
purpose of minibatch.shuffle() in training loop

### ISSUE THREAD

**Issue by abasak24** [GraphSAINT/GraphSAINT#22]
Hi, 
In the training loop, I notice there is a **minibatch.shuffle()**  at the beginning of each epoch. Upon inspecting minibatch.py, I observe that this function performs **np.random.permutation(self.node_train)**. Actually, it is not clear to me what this shuffling is doing and what is its benefit. Could you please clarify?

**Comment by ZimpleX** [comment]
This function is shuffling the indices of the training nodes. In normal deep learning setup, we want stochastic behavior in each epoch. People will do this kind of shuffling so that minibatches across different epochs contain different data points.  

In the case of GraphSAINT, you are correct that this shuffling is not necessary. The reason is that the subgraphs are randomly sampled from the training graph anyways. So whether or not we have this shuffling, there shouldn't be any difference in terms of convergence. 

So you can either remove it or leave it as it is.

**Comment by abasak24** [comment]
Thank you for clarifying!
